src/guiapi/src/window.cpp

```diff
@@ -67,19 +71,42 @@ void window_t::SetCapture() {
 }
 
 void window_t::Show() {
-    if (!IsVisible()) {
+    if (!flag_visible) {
         flag_visible = true;
-        Invalidate();
+        //cannot invalidate when is hidden by dialog - could flicker
+        if (!flag_hidden_behind_dialog)
+            Invalidate();
     }
 }
 
 void window_t::Hide() {
-    if (IsVisible()) {
+    if (flag_visible) {
         flag_visible = false;
+        //cannot invalidate when is hidden by dialog - could flicker
+        if (!flag_hidden_behind_dialog)
+            Invalidate();
+    }
+}
+
+void window_t::ShowAfterDialog() {
+    if (flag_hidden_behind_dialog) {
+        flag_hidden_behind_dialog = false;
+        //must invalidate even when is not visible
         Invalidate();
     }
 }
 
+void window_t::HideBehindDialog() {
```

Comment:
ShowAfterDialog and HideBehindDialog doing the same stuff. Is this work in progress to invalidate those hidden parts which intersect with dialog rect ?

Reply:
It is inversive operation

Reply:
oh it ha same code, its a bug
thx